modules/codegen/test/src/smithy4s/codegen/internals/RendererSpec.scala

```diff
@@ -85,6 +85,192 @@ final class RendererSpec extends munit.FunSuite {
       s"""val underlyingSchema: Schema[Map[String, Int]] = map($keySchemaString, $valueSchemaString)"""
     assert(definition.contains(requiredString))
   }
+  test("enum trait hints should be preserved") {
+    val smithy = s"""
+                    |$$version: "2.0"
+                    |namespace smithy4s
+                    |
+                    |@documentation("this is an enum Suit")
+                    |@enum([
+                    |{
+                    |  value: "DIAMOND",
+                    |  name: "DIAMOND",
+                    |  documentation: "this is a DIAMOND"
+                    |},
+                    |{
+                    |  value: "HAERT",
+                    |  name: "HAERT",
+                    |  documentation: "this is a HAERT",
+                    |  deprecated: true
+                    |},
+                    |{
+                    |  value: "HEART",
+                    |  name: "HEART",
+                    |  documentation: "this is a HEART"
+                    |},
+                    |{
+                    |  value: "CLUB",
+                    |  name: "CLUB"
+                    |},
+                    |{
+                    |  value: "SPADE"
+                    |  name: "SPADE"
+                    |}
+                    |])
+                    |string Suit
+                    |
+                    |""".stripMargin
+    val contents = generateScalaCode(smithy).values
+    val definition =
+      contents.find(content =>
+        content.contains("sealed abstract class Suit") && content.contains(
+          "object Suit extends Enumeration[Suit]"
+        )
+      ) match {
+        case None =>
+          fail(
+            "enum must be rendered as a sealed abstract class and its companion object"
+          )
+        case Some(code) => code
+      }
+    val classDoc = """/** this is an enum Suit
+                     |  * @param DIAMOND
+                     |  *   this is a DIAMOND
+                     |  * @param HAERT
+                     |  *   this is a HAERT
+                     |  * @param HEART
+                     |  *   this is a HEART
+                     |  */""".stripMargin
+    val diamondWithDocRendered =
+      """case object DIAMOND extends Suit("DIAMOND", "DIAMOND", 0, Hints(smithy.api.Documentation("this is a DIAMOND")))"""
+    // NOTE: enum trait and enum shape are not 100% compatible. For example, enum trait doesn't support deprecated$since and deprecated$message.
+    val haertWithDeprecationAndDocRendered = List(
```

Comment:
Here, I use List instead of single String because multiline assertion somehow failed on CI but worked on my local machine.